src/main/kotlin/org/jitsi/rtp/rtp/RtpPacket.kt

```diff
@@ -224,11 +224,14 @@ open class RtpPacket(
 
         // Copy the header (and extensions, excluding padding, if there are any) to
         // the very beginning of the buffer
-        System.arraycopy(buffer, offset,
-            newBuffer, 0,
-            newHeaderLength)
+        if (buffer != newBuffer || offset != 0) {
```

Comment:
wait, even if we're using the same buffer, we still may want to move the data, no?

Reply:
is this because we don't generally share buffers across rtp packets?

Reply:
@bbaldino, the `offset != 0` (0 is the dst offset) check takes care of that.

Reply:
@gpolitis this is just to avoid calling `arraycopy(buffer, 0, buffer, 0, len)`, because it wouldn't have any effect (but my tests show that it will perform the copy, wasting cycles)

Reply:
i see now, i misread.